src/main/kotlin/com/coder/gateway/CoderGatewayConnectionProvider.kt

```diff
@@ -93,31 +109,38 @@ class CoderGatewayConnectionProvider : GatewayConnectionProvider {
             cli.login(client.token)
 
             indicator.text = "Configuring Coder CLI..."
-            cli.configSsh(client.agentNames(workspaces))
-
-            // TODO: Ask for these if missing.  Maybe we can reuse the second
-            //  step of the wizard?  Could also be nice if we automatically used
-            //  the last IDE.
-            if (parameters[IDE_PRODUCT_CODE].isNullOrBlank()) {
-                throw IllegalArgumentException("Query parameter \"$IDE_PRODUCT_CODE\" is missing")
-            }
-            if (parameters[IDE_BUILD_NUMBER].isNullOrBlank()) {
-                throw IllegalArgumentException("Query parameter \"$IDE_BUILD_NUMBER\" is missing")
-            }
-            if (parameters[IDE_PATH_ON_HOST].isNullOrBlank() && parameters[IDE_DOWNLOAD_LINK].isNullOrBlank()) {
-                throw IllegalArgumentException("One of \"$IDE_PATH_ON_HOST\" or \"$IDE_DOWNLOAD_LINK\" is required")
-            }
-
-            // Check that both the domain and the redirected domain are
-            // allowlisted.  If not, check with the user whether to proceed.
-            verifyDownloadLink(parameters)
-
-            // TODO: Ask for the project path if missing and validate the path.
-            val folder = parameters[FOLDER] ?: throw IllegalArgumentException("Query parameter \"$FOLDER\" is missing")
-
-            parameters
-                .withWorkspaceHostname(CoderCLIManager.getHostName(deploymentURL.toURL(), agent.name))
-                .withProjectPath(folder)
+            cli.configSsh(client.agentNames(workspaces).toSet())
+
+
+            val openDialog = if (parameters[IDE_PRODUCT_CODE].isNullOrBlank())
+                true
+            else if (parameters[IDE_BUILD_NUMBER].isNullOrBlank())
+                true
+            else if (parameters[IDE_PATH_ON_HOST].isNullOrBlank() && parameters[IDE_DOWNLOAD_LINK].isNullOrBlank())
+                true
+            else if (parameters[FOLDER].isNullOrBlank())
+                true
+            else
+                false
+
+            val params = if (openDialog) {
+                val view = CoderWorkspaceStepView{}
+                ApplicationManager.getApplication().invokeAndWait {
+                    view.init(
+                        CoderWorkspacesStepSelection(agent, workspace, cli, client, workspaces)
+                    )
+                    val dialog = SelectWorkspaceIDEDialog(view.component)
+                    dialog.show()
+                }
+                val p = parameters.toMutableMap()
+                listOf(IDE_PRODUCT_CODE, IDE_BUILD_NUMBER, IDE_DOWNLOAD_LINK, IDE_PATH_ON_HOST, PROJECT_PATH).forEach { prop ->
+                    view.data()[prop]?.let { value -> p[prop] = value }
+                }
+                p
```

Comment:
So maybe we prefer `paramaters` and not overwrite with what we get back from the View.  Then if one of `IDE_PATH_ON_HOST` or `IDE_DOWNLOAD_LINK` is set in params we skip setting those?

Reply:
That works for me!  I think at some point I want to refactor that step to only return IDE data instead of the entire parameter set anyway, so this will be good in preparation for that.

Reply:
Wait I think I misunderstood, if we ignore the path or host from the view and use whatever is in the original parameters we could still get a mismatch.  If the original parameter link is `http://jetbrains-221.tar.gz` for example but in the step we selected `232` or some other version.

Basically to be safe I think we have to always use all the IDE-related params from the original, or all the IDE-related params from the step, but never a combination of them.

Reply:
okay, makes sense, I put it back so we will prefer `IDE_PRODUCT_CODE, IDE_BUILD_NUMBER, PROJECT_PATH, IDE_PATH_ON_HOST, IDE_DOWNLOAD_LINK` returned from the view.

Reply:
Thank you, I think one of path or link from the step could be null which means we would end up using the value in the original parameters, but let me test that out when I add my changes and I will adjust if we need to!